schemas/mozilla_nimbus_schemas/experiments/experiments.py

```diff
@@ -273,3 +294,50 @@ class NimbusExperiment(BaseModel):
         ),
         default=None,
     )
+
+    @model_validator(mode="after")
+    @classmethod
+    def validate_firefox_labs(cls, data: Self) -> Self:
+        if data.isFirefoxLabsOptIn:
+            if data.firefoxLabsTitle is None:
+                raise ValueError(
+                    "firefoxLabsTitle field cannot be none "
+                    "if isFirefoxLabsOptIn is True"
+                )
+            if data.firefoxLabsDescription is None:
+                raise ValueError(
+                    "firefoxLabsDescription field cannot be none "
+                    "if isFirefoxLabsOptIn is True"
+                )
+            if not data.isRollout:
+                for branch in data.branches:
+                    if branch.firefoxLabsTitle is None:
+                        raise ValueError(f"{branch.slug} missing firefoxLabsTitle")
```

Comment:
```suggestion
                        raise ValueError(f"branch with slug {branch.slug} is missing firefoxLabsTitle field (required because firefoxLabsTitle is True)")
```